blog/05-03-23-deploy-and-interact-with-smart-contracts-on-celo/deploy-and-interact-with-smart-contracts-on-celo.md

```diff
@@ -0,0 +1,250 @@
+The following tools are what you need to work with to deploy and operate smart contracts on the Celo blockchain:
+
+
+1. Celo CLI
+2. A text editor (e.g., Visual Studio Code)
+3. Node.js
+4. npm (Node Package Manager)
+5. Ganache CLI (for local development)
+6. Solidity compiler (e.g., solc)
+
+# Steps to Deploying and Interacting with Smart Contracts on Celo Blockchain
+
+
+1. Setting up a Local Development Environment
+
+A development environment is an environment in which all resources and tools used to develop a program or software product are available. The first step to deploying is an attempt to create a development environment with a set of processes and tools used to develop [smart contracts](https://www.coinbase.com/learn/crypto-basics/what-is-a-smart-contract).
+
+You can set up your local development environment by following these steps:
+
+Step 1: Install Node.js and npm on your machine.
+
+Step 2: Install the Celo CLI by running the following command in a terminal.
+
+npm install -g @celo/cli
+
+Step 3: Install the Ganache CLI by running the following command in your terminal.
+
+npm install -g ganache- cli
+
+Step 4. Install the Solidity compiler by running the following command in Terminal:
+
+npm install -g solc
+
+
+2. Writing a Smart Contract
+
+You can write a smart contract by following these steps:
+
+Step 1: Open your text editor and create a new file.
+
+Step 2: Write the code for your smart contract in Solidity.
+
+For example, you could write a simple smart contract that stores a string:
+
+**C#>>**
+
+```
+  ```pragma solidity ^0.8.0;
+
+contract MyContract {
+    string public myString;
+
+    function setMyString(string memory _myString) public {
+        myString = _myString;
+    }
+}```
+```
+
+Step 3: Save the file with a .sol extension (e.g., MyContract.sol).
+
+
+3. Compiling the Smart Contract
+
+After creating file extensions for operation purposes e.g saving strings, compile the smart contract by following these steps:
+
+Step 1: Open your terminal and navigate to the smart contract file's directory.
+
+Step 2: Run the following command to compile the smart contract:
+
+**CSS>>** 
+
+ 
+
+```
+ solc MyContract.sol --bin --abi --optimize -o ./build
+```
+
+This command compiles the smart contract and generates two files: 
+
+
+* MyContract.bin, and 
+* MyContract.abi. 
+
+The .bin file contains the bytecode of the compiled smart contract, and the .abi file contains the Application Binary Interface (ABI) that describes how to interact with the smart contract.
+
+
+4. Deploying the Smart Contract
+
+You can [deploy the smart contract on the Celo blockchain](https://docs.celo.org/community/release-process/smart-contracts) following these steps:
+
+Step 1: Start the Ganache CLI by running the following command in your terminal:
+
+ganache-cli
+
+Step 2: Open a new terminal window and navigate to the smart contract file's directory.
+
+Step 3: Run the following command to deploy the smart contract:
+
+**Python>>**
```

Comment:
if I am not wrong you adding this for showing what kind of code it is? you can just use ```bash if its a command ```solidity if its a solidity code ```js for javascript

Reply:
Alright. Will make the edits where necessary